libs/ml/tests/unit/ops/avg_pool_1d.cpp

```diff
@@ -131,48 +137,48 @@ TYPED_TEST(AvgPool1DTest, backward_test_2_channels)
   using TensorType = TypeParam;
   using SizeType   = fetch::math::SizeType;
 
-  TensorType          data({2, 5, 2});
-  TensorType          error({2, 2, 2});
-  TensorType          gt({2, 5, 2});
-  std::vector<double> data_input({1, -2, 3, -4, 10, -6, 7, -8, 9, -10});
-  std::vector<double> errorInput({2, 3, 4, 5});
+  TensorType data({2, 5, 2});
+  TensorType error({2, 2, 2});
+  TensorType gt({2, 5, 2});
+  TensorType data_input  = TensorType::FromString("1, -2, 3, -4, 10, -6, 7, -8, 9, -10");
+  TensorType error_input = TensorType::FromString("2, 3, 4, 5");
 
   for (SizeType i{0}; i < 2; ++i)
   {
     for (SizeType j{0}; j < 5; ++j)
     {
-      data(i, j, 0) = static_cast<DataType>(data_input[i * 5 + j]);
+      data(i, j, 0) = data_input[i * 5 + j];
     }
   }
 
   for (SizeType i{0}; i < 2; ++i)
   {
     for (SizeType j{0}; j < 2; ++j)
     {
-      error(i, j, 0) = static_cast<DataType>(errorInput[i * 2 + j]);
+      error(i, j, 0) = error_input[i * 2 + j];
     }
   }
 
-  gt(0, 0, 0) = DataType{0.5f};
-  gt(0, 0, 1) = DataType{0.0f};
-  gt(0, 1, 0) = DataType{1.25f};
-  gt(0, 1, 1) = DataType{0.0f};
-  gt(0, 2, 0) = DataType{1.25f};
-  gt(0, 2, 1) = DataType{0.0f};
-  gt(0, 3, 0) = DataType{1.25f};
-  gt(0, 3, 1) = DataType{0.0f};
-  gt(0, 4, 0) = DataType{0.75f};
-  gt(0, 4, 1) = DataType{0.0f};
-  gt(1, 0, 0) = DataType{1.0f};
-  gt(1, 0, 1) = DataType{0.0f};
-  gt(1, 1, 0) = DataType{2.25f};
-  gt(1, 1, 1) = DataType{0.0f};
-  gt(1, 2, 0) = DataType{2.25f};
-  gt(1, 2, 1) = DataType{0.0f};
-  gt(1, 3, 0) = DataType{2.25f};
-  gt(1, 3, 1) = DataType{0.0f};
-  gt(1, 4, 0) = DataType{1.25f};
-  gt(1, 4, 1) = DataType{0.0f};
+  gt(0, 0, 0) = DataType{1} / DataType{2};
+  gt(0, 0, 1) = DataType{0};
+  gt(0, 1, 0) = DataType{5} / DataType{4};
```

Comment:
well, division of an integer by 2 or by 4 would result in xx.5 or xx.25 floats, which could be represented in fixed point with 100% accuracy, so DataType{5 / 4} is the just same (in this particular case!), right?